### DISCUSSION TITLE
How do you update OTA?

### DISCUSSION THREAD

**Discussion by mbdawson** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#184] [Q&A]
Hello. Another successful install for a green coleman egg pump. I installed a few days ago and enabled OTA updates after the initial upload and was able to successfully upload OTA. I noticed now there is a new build out and I was curious how do I go about updating the device OTA?

How do I go about loading the new code into Visual Studio and uploading it?

**Comment by visualapproach** [comment]
Hello, just download the code and open in VSC. Edit "platformio.ini" with your device IP
```
upload_protocol = espota
upload_port = 192.168.4.121
upload_flags = 
	--auth=esp8266
```
save and press upload (the right arrow icon at the bottom).

**Comment by dozwin81** [comment]
[Update OTA.pdf](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/files/13181773/Update.OTA.pdf)

I cant Upload, must i change anything in the platformio.ini, i have change your lines!?

**Reply by visualapproach** [reply]
Hard to tell from the pdf. Maybe wrong indentation, but this line
`Serial port /dev/cu.Bluetooth-Incoming-Port `
seems weird. Guess you have a mac and I don't know how it works, but blutooth incoming port seems wrong. It tries to upload via serial.

Use ctrl+' (I guess cmd+') to toggle comments. That way you will keep the correct indent.

**Reply by dozwin81** [reply]
I have change to a win 11 Laptop and have the next Problems!

**Platformio.ini:**

; PlatformIO Project Configuration File
;
;   Build options: build flags, source filter
;   Upload options: custom upload port, speed and extra flags
;   Library options: dependencies, extra library storages
;   Advanced options: extra scripting
;
; Please visit documentation for the other options and examples
; https://docs.platformio.org/page/projectconf.html

[platformio]
src_dir = src
data_dir = datazip
; default_envs = esp32

[env:nodemcuv2]
platform = espressif8266@^4
board = nodemcuv2
framework = arduino
lib_deps = 
	bblanchon/ArduinoJson
	; mcxiaoke/ESPDateTime
	links2004/WebSockets
	knolleary/PubSubClient
	me-no-dev/ESPAsyncTCP
	; khoih-prog/ESP_WifiManager@^1.12.1
    tzapu/WifiManager
	plerup/EspSoftwareSerial
	milesburton/DallasTemperature
board_build.filesystem = littlefs
extra_scripts = gzip_littlefs.py
monitor_speed = 115200
board_build.f_cpu = 160000000L
; board_build.f_flash = 80000000L;
upload_speed = 921600
build_flags = 
	-DWEBSOCKETS_NETWORK_TYPE="NETWORK_ESP8266_ASYNC"
    -DWEBSOCKETS_SAVE_RAM
    ; -DDEBUG_ESP_PORT=Serial
    ; -D BUILDTIME=$UNIX_TIME
    ; -DDEBUG_ESP_SSL
    ; -DDEBUG_ESP_TLS_MEM
    ; -DDEBUG_ESPHTTP_CLIENT
    ; -DDEBUG_ESP_HTTP_UPDATE
    ; -DDEBUG_ESP_OOM
    ; -DDEBUG_ESP_MMU
    ; -include "umm_malloc/umm_malloc_cfg.h"
	; -DPIO_FRAMEWORK_ARDUINO_MMU_CACHE16_IRAM48
    -D PIO_FRAMEWORK_ARDUINO_MMU_CACHE16_IRAM48_SECHEAP_SHARED
    ; -D PIO_FRAMEWORK_ARDUINO_MMU_CACHE16_IRAM32_SECHEAP_NOTSHARED
    ; -D PIO_FRAMEWORK_ARDUINO_MMU_CUSTOM
    ; -D MMU_IRAM_SIZE=0xC000
    ; -D MMU_ICACHE_SIZE=0x4000
    ; -D MMU_IRAM_HEAP    
    -DVTABLES_IN_FLASH
    ; -DPIO_FRAMEWORK_ARDUINO_LWIP2_HIGHER_BANDWIDTH
    ; -D PIO_FRAMEWORK_ARDUINO_LWIP2_LOW_MEMORY
    ; -D BEARSSL_SSL_BASIC
; build_type = debug
; debug_build_flags = -Og -g2 -ggdb2
monitor_filters = 
	esp8266_exception_decoder
	default
    ; log2file
; upload_protocol = esptool
upload_protocol = espota
upload_port = 192.168.2.59
upload_flags = 
--auth=esp8266


; [env:esp32]
; platform = espressif32
; board = esp32doit-devkit-v1
; framework = arduino
; board_build.filesystem = littlefs
; upload_speed = 921600
; monitor_speed = 115200
; lib_deps = 
; 	bblanchon/ArduinoJson @ ^6.20.0
; 	mcxiaoke/ESPDateTime @ ^1.0.4
; 	links2004/WebSockets @ ^2.3.7
; 	knolleary/PubSubClient @ ^2.8
; 	me-no-dev/ESPAsyncTCP @ ^1.2.2
; 	khoih-prog/ESP_WifiManager@^1.12.1
; 	plerup/EspSoftwareSerial@^7.0.0
; upload_protocol = esptool
; ;upload_protocol = espota
; ;upload_port = layzspa.local
; ;upload_flags = 
; ;	--auth=esp8266

**Problem:**

[{
	"resource": "/c:/Users/bjoer/Downloads/WiFi-remote-for-Bestway-Lay-Z-SPA-4.2.3/WiFi-remote-for-Bestway-Lay-Z-SPA-4.2.3/Code/platformio.ini",
	"owner": "PlatformIO",
	"severity": 4,
	"message": "Ignore unknown configuration option `--auth` in section [env:nodemcuv2]",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}]

**Build:**

Executing task: C:\Users\bjoer\.platformio\penv\Scripts\platformio.exe run 

Warning! Ignore unknown configuration option `--auth` in section [env:nodemcuv2]
Processing nodemcuv2 (platform: espressif8266@^4; board: nodemcuv2; framework: arduino)
------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/nodemcuv2.html
PLATFORM: Espressif 8266 (4.2.1) > NodeMCU 1.0 (ESP-12E Module)
HARDWARE: ESP8266 160MHz, 80KB RAM, 4MB Flash
PACKAGES: 
 - framework-arduinoespressif8266 @ 3.30102.0 (3.1.2) 
 - tool-esptool @ 1.413.0 (4.13) 
 - tool-esptoolpy @ 1.30000.201119 (3.0.0)
 - toolchain-xtensa @ 2.100300.220621 (10.3.0)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 48 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ArduinoJson @ 6.21.3
|-- WebSockets @ 2.4.1
|-- PubSubClient @ 2.8.0
|-- ESPAsyncTCP @ 1.2.2
|-- WiFiManager @ 0.16.0
|-- EspSoftwareSerial @ 8.1.0
|-- DallasTemperature @ 3.11.0
|-- BWC_unified
|-- ArduinoOTA @ 1.0
|-- DNSServer @ 1.1.1
|-- ESP8266HTTPClient @ 1.2
|-- ESP8266httpUpdate @ 1.3
|-- ESP8266WebServer @ 1.0
|-- ESP8266WiFi @ 1.0
|-- LittleFS @ 0.1.0
|-- OneWire @ 2.3.7
|-- Ticker @ 1.0
Building in release mode
Compiling .pio\build\nodemcuv2\src\main.cpp.o
Generating LD script .pio\build\nodemcuv2\ld\local.eagle.app.v6.common.ld
Compiling .pio\build\nodemcuv2\libb5e\Hash\Hash.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\BearSSLHelpers.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\CertStoreBearSSL.cpp.o
Archiving .pio\build\nodemcuv2\libb5e\libHash.a
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\ESP8266WiFi.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\ESP8266WiFiAP.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\ESP8266WiFiGeneric.cpp.o
src\main.cpp: In function 'void stopall()':
src\main.cpp:509:5: warning: deleting object of polymorphic class type 'PubSubClient' which has non-virtual destructor might cause undefined behavior [-Wdelete-non-virtual-dtor]
  509 |     delete mqttClient;
      |     ^~~~~~~~~~~~~~~~~
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\ESP8266WiFiGratuitous.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\ESP8266WiFiMulti.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\ESP8266WiFiSTA-WPS.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\ESP8266WiFiSTA.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\ESP8266WiFiScan.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\WiFiClient.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\WiFiClientSecureBearSSL.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\WiFiServer.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\WiFiServerSecureBearSSL.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\WiFiUdp.cpp.o
Compiling .pio\build\nodemcuv2\libaf4\ESP8266WiFi\enable_wifi_at_boot_time.cpp.o
Compiling .pio\build\nodemcuv2\lib374\ESPAsyncTCP\AsyncPrinter.cpp.o
Compiling .pio\build\nodemcuv2\lib374\ESPAsyncTCP\ESPAsyncTCP.cpp.o
Archiving .pio\build\nodemcuv2\libaf4\libESP8266WiFi.a
Compiling .pio\build\nodemcuv2\lib374\ESPAsyncTCP\ESPAsyncTCPbuffer.cpp.o
Compiling .pio\build\nodemcuv2\lib374\ESPAsyncTCP\SyncClient.cpp.o
Compiling .pio\build\nodemcuv2\lib374\ESPAsyncTCP\tcp_axtls.c.o
Compiling .pio\build\nodemcuv2\lib6f8\SPI\SPI.cpp.o
Compiling .pio\build\nodemcuv2\liba58\Ethernet\Dhcp.cpp.o
Archiving .pio\build\nodemcuv2\lib374\libESPAsyncTCP.a
Archiving .pio\build\nodemcuv2\lib6f8\libSPI.a
Compiling .pio\build\nodemcuv2\liba58\Ethernet\Dns.cpp.o
Compiling .pio\build\nodemcuv2\liba58\Ethernet\Ethernet.cpp.o
Compiling .pio\build\nodemcuv2\liba58\Ethernet\EthernetClient.cpp.o
Compiling .pio\build\nodemcuv2\liba58\Ethernet\EthernetServer.cpp.o
Compiling .pio\build\nodemcuv2\liba58\Ethernet\EthernetUdp.cpp.o
Compiling .pio\build\nodemcuv2\liba58\Ethernet\socket.cpp.o
Compiling .pio\build\nodemcuv2\liba58\Ethernet\utility\w5100.cpp.o
Compiling .pio\build\nodemcuv2\lib3e2\WebSockets\SocketIOclient.cpp.o
Compiling .pio\build\nodemcuv2\lib3e2\WebSockets\WebSockets.cpp.o
.pio\libdeps\nodemcuv2\WebSockets\src\SocketIOclient.cpp: In member function 'void SocketIOclient::loop()':
.pio\libdeps\nodemcuv2\WebSockets\src\SocketIOclient.cpp:175:23: warning: 'void WebSocketsClient::loop()' is deprecated [-Wdeprecated-declarations]
  175 |     WebSocketsClient::loop();
      |                       ^~~~
In file included from .pio\libdeps\nodemcuv2\WebSockets\src\SocketIOclient.cpp:9:
.pio\libdeps\nodemcuv2\WebSockets\src\WebSocketsClient.h:75:10: note: declared here
   75 |     void loop(void) __attribute__((deprecated)) {}
      |          ^~~~
.pio\libdeps\nodemcuv2\WebSockets\src\SocketIOclient.cpp:175:28: warning: 'void WebSocketsClient::loop()' is deprecated [-Wdeprecated-declarations]
  175 |     WebSocketsClient::loop();
      |                            ^
In file included from .pio\libdeps\nodemcuv2\WebSockets\src\SocketIOclient.cpp:9:
.pio\libdeps\nodemcuv2\WebSockets\src\WebSocketsClient.h:75:10: note: declared here
   75 |     void loop(void) __attribute__((deprecated)) {}
      |          ^~~~
Archiving .pio\build\nodemcuv2\liba58\libEthernet.a
Compiling .pio\build\nodemcuv2\lib3e2\WebSockets\WebSocketsClient.cpp.o
Compiling .pio\build\nodemcuv2\lib3e2\WebSockets\WebSocketsServer.cpp.o
Compiling .pio\build\nodemcuv2\lib3e2\WebSockets\libb64\cdecode.c.o
Compiling .pio\build\nodemcuv2\lib3e2\WebSockets\libb64\cencode.c.o
Compiling .pio\build\nodemcuv2\lib3e2\WebSockets\libsha1\libsha1.c.o
Compiling .pio\build\nodemcuv2\lib040\PubSubClient\PubSubClient.cpp.o
Compiling .pio\build\nodemcuv2\lib161\DNSServer\DNSServer.cpp.o
Archiving .pio\build\nodemcuv2\lib3e2\libWebSockets.a
Compiling .pio\build\nodemcuv2\lib760\ESP8266WebServer\detail\mimetable.cpp.o
.pio\libdeps\nodemcuv2\PubSubClient\src\PubSubClient.cpp: In member function 'boolean PubSubClient::publish_P(const char*, const uint8_t*, unsigned int, boolean)':
.pio\libdeps\nodemcuv2\PubSubClient\src\PubSubClient.cpp:523:16: warning: comparison of integer expressions of different signedness: 'unsigned int' and 'int' [-Wsign-compare]
  523 |     return (rc == expectedLength);
      |             ~~~^~~~~~~~~~~~~~~~~
Compiling .pio\build\nodemcuv2\libc67\WiFiManager\WiFiManager.cpp.o
Archiving .pio\build\nodemcuv2\lib760\libESP8266WebServer.a
Archiving .pio\build\nodemcuv2\lib040\libPubSubClient.a
Compiling .pio\build\nodemcuv2\libf39\EspSoftwareSerial\SoftwareSerial.cpp.o
Archiving .pio\build\nodemcuv2\lib161\libDNSServer.a
Compiling .pio\build\nodemcuv2\lib278\OneWire\OneWire.cpp.o
Compiling .pio\build\nodemcuv2\lib31a\DallasTemperature\DallasTemperature.cpp.o
.pio\libdeps\nodemcuv2\OneWire\OneWire.cpp: In member function 'uint8_t OneWire::reset()':
.pio\libdeps\nodemcuv2\OneWire\OneWire.cpp:167:24: warning: unused variable 'reg' [-Wunused-variable]
  167 |  volatile IO_REG_TYPE *reg IO_REG_BASE_ATTR = baseReg;
      |                        ^~~
.pio\libdeps\nodemcuv2\OneWire\OneWire.cpp: In member function 'void OneWire::write_bit(uint8_t)':
.pio\libdeps\nodemcuv2\OneWire\OneWire.cpp:201:24: warning: unused variable 'reg' [-Wunused-variable]
  201 |  volatile IO_REG_TYPE *reg IO_REG_BASE_ATTR = baseReg;
      |                        ^~~
.pio\libdeps\nodemcuv2\OneWire\OneWire.cpp: In member function 'uint8_t OneWire::read_bit()':
.pio\libdeps\nodemcuv2\OneWire\OneWire.cpp:229:24: warning: unused variable 'reg' [-Wunused-variable]
  229 |  volatile IO_REG_TYPE *reg IO_REG_BASE_ATTR = baseReg;
      |                        ^~~
Archiving .pio\build\nodemcuv2\lib278\libOneWire.a
Archiving .pio\build\nodemcuv2\libf39\libEspSoftwareSerial.a
Archiving .pio\build\nodemcuv2\lib31a\libDallasTemperature.a
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_2021.cpp.o
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_2021HJT.cpp.o
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_4W.cpp.o
Archiving .pio\build\nodemcuv2\libc67\libWiFiManager.a
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_54149E.cpp.o
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_6W.cpp.o
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_BASE.cpp.o
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_PRE2021.cpp.o
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_TYPE1.cpp.o
Compiling .pio\build\nodemcuv2\libddd\cio\CIO_TYPE2.cpp.o
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_2021.cpp.o
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_2021HJT.cpp.o
Archiving .pio\build\nodemcuv2\libddd\libcio.a
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_4W.cpp.o
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_54149E.cpp.o
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_6W.cpp.o
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_BASE.cpp.o
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_PRE2021.cpp.o
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_TYPE1.cpp.o
Compiling .pio\build\nodemcuv2\libf39\dsp\DSP_TYPE2.cpp.o
Compiling .pio\build\nodemcuv2\lib54a\LittleFS\LittleFS.cpp.o
Compiling .pio\build\nodemcuv2\lib54a\LittleFS\lfs.c.o
Compiling .pio\build\nodemcuv2\lib54a\LittleFS\lfs_util.c.o
Archiving .pio\build\nodemcuv2\libf39\libdsp.a
Compiling .pio\build\nodemcuv2\libd0a\Ticker\Ticker.cpp.o
Compiling .pio\build\nodemcuv2\lib9b8\BWC_unified\bwc.cpp.o
Compiling .pio\build\nodemcuv2\lib9b8\BWC_unified\util.cpp.o
Compiling .pio\build\nodemcuv2\lib424\ESP8266mDNS\ESP8266mDNS.cpp.o
Archiving .pio\build\nodemcuv2\lib54a\libLittleFS.a
Compiling .pio\build\nodemcuv2\lib424\ESP8266mDNS\LEAmDNS.cpp.o
Archiving .pio\build\nodemcuv2\libd0a\libTicker.a
Compiling .pio\build\nodemcuv2\lib424\ESP8266mDNS\LEAmDNS_Control.cpp.o
Compiling .pio\build\nodemcuv2\lib424\ESP8266mDNS\LEAmDNS_Helpers.cpp.o
Compiling .pio\build\nodemcuv2\lib424\ESP8266mDNS\LEAmDNS_Structs.cpp.o
Compiling .pio\build\nodemcuv2\lib424\ESP8266mDNS\LEAmDNS_Transfer.cpp.o
Compiling .pio\build\nodemcuv2\lib6f8\ArduinoOTA\ArduinoOTA.cpp.o
Archiving .pio\build\nodemcuv2\lib9b8\libBWC_unified.a
Compiling .pio\build\nodemcuv2\lib8af\ESP8266HTTPClient\ESP8266HTTPClient.cpp.o
Compiling .pio\build\nodemcuv2\lib68a\ESP8266httpUpdate\ESP8266httpUpdate.cpp.o
Archiving .pio\build\nodemcuv2\lib424\libESP8266mDNS.a
Archiving .pio\build\nodemcuv2\libFrameworkArduinoVariant.a
Compiling .pio\build\nodemcuv2\FrameworkArduino\Crypto.cpp.o
Archiving .pio\build\nodemcuv2\lib6f8\libArduinoOTA.a
Compiling .pio\build\nodemcuv2\FrameworkArduino\Esp-frag.cpp.o
Archiving .pio\build\nodemcuv2\lib68a\libESP8266httpUpdate.a
Compiling .pio\build\nodemcuv2\FrameworkArduino\Esp-version.cpp.o
Archiving .pio\build\nodemcuv2\lib8af\libESP8266HTTPClient.a
Compiling .pio\build\nodemcuv2\FrameworkArduino\Esp.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\FS.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\FSnoop.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\FunctionalInterrupt.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\HardwareSerial.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\IPAddress.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\LwipDhcpServer-NonOS.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\LwipDhcpServer.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\LwipIntf.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\LwipIntfCB.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\MD5Builder.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\Print.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\Schedule.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\StackThunk.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\Stream.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\StreamSend.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\Tone.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\TypeConversion.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\Updater.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\WMath.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\WString.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\abi.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\aes_unwrap.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\base64.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\cbuf.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\cont.S.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\cont_util.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_app_entry_noextra4k.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_eboot_command.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_features.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_flash_quirks.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_flash_utils.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_i2s.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_main.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_non32xfer.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_noniso.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_phy.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_postmortem.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_si2c.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_sigma_delta.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_spi_utils.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_timer.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_vm.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_waveform_phase.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_waveform_pwm.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_wiring.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_wiring_analog.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_wiring_digital.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_wiring_pulse.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_wiring_pwm.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\core_esp8266_wiring_shift.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\crc32.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\debug.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\exc-c-wrapper-handler.S.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\exc-sethandler.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\flash_hal.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\gdb_hooks.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\heap.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\hwdt_app_entry.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\libb64\cdecode.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\libb64\cencode.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\libc_replacements.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\mmu_iram.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\reboot_uart_dwnld.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\spiffs\spiffs_cache.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\spiffs\spiffs_check.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\spiffs\spiffs_gc.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\spiffs\spiffs_hydrogen.cpp.o
C:\Users\bjoer\.platformio\packages\framework-arduinoespressif8266\cores\esp8266\mmu_iram.cpp:29:34: note: '#pragma message: ICACHE size 16K'
   29 | #pragma message("ICACHE size 16K")
      |                                  ^
Compiling .pio\build\nodemcuv2\FrameworkArduino\spiffs\spiffs_nucleus.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\spiffs_api.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\sqrt32.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\stdlib_noniso.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\time.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\uart.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\umm_malloc\umm_info.c.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\umm_malloc\umm_integrity.c.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\umm_malloc\umm_local.c.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\umm_malloc\umm_malloc.cpp.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\umm_malloc\umm_poison.c.o
Compiling .pio\build\nodemcuv2\FrameworkArduino\wpa2_eap_patch.cpp.o
Archiving .pio\build\nodemcuv2\libFrameworkArduino.a
Linking .pio\build\nodemcuv2\firmware.elf
Retrieving maximum program size .pio\build\nodemcuv2\firmware.elf
Checking size .pio\build\nodemcuv2\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [=====     ]  49.6% (used 40644 bytes from 81920 bytes)
Flash: [=====     ]  53.2% (used 556119 bytes from 1044464 bytes)
Building .pio\build\nodemcuv2\firmware.bin
Creating BIN file ".pio\build\nodemcuv2\firmware.bin" using "C:\Users\bjoer\.platformio\packages\framework-arduinoespressif8266\bootloaders\eboot\eboot.elf" and ".pio\build\nodemcuv2\firmware.elf"
================================================== [SUCCESS] Took 176.86 seconds ==================================================
 *  Terminal will be reused by tasks, press any key to close it. 

**Upload:**

 *  Executing task: C:\Users\bjoer\.platformio\penv\Scripts\platformio.exe run --target upload 

Warning! Ignore unknown configuration option `--auth` in section [env:nodemcuv2]
Processing nodemcuv2 (platform: espressif8266@^4; board: nodemcuv2; framework: arduino)
------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/nodemcuv2.html
PLATFORM: Espressif 8266 (4.2.1) > NodeMCU 1.0 (ESP-12E Module)
HARDWARE: ESP8266 160MHz, 80KB RAM, 4MB Flash
PACKAGES: 
 - framework-arduinoespressif8266 @ 3.30102.0 (3.1.2) 
 - tool-esptool @ 1.413.0 (4.13) 
 - tool-esptoolpy @ 1.30000.201119 (3.0.0)
 - tool-mklittlefs @ 1.203.210628 (2.3)
 - tool-mkspiffs @ 1.200.0 (2.0)
 - toolchain-xtensa @ 2.100300.220621 (10.3.0)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 48 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ArduinoJson @ 6.21.3
|-- WebSockets @ 2.4.1
|-- PubSubClient @ 2.8.0
|-- ESPAsyncTCP @ 1.2.2
|-- WiFiManager @ 0.16.0
|-- EspSoftwareSerial @ 8.1.0
|-- DallasTemperature @ 3.11.0
|-- BWC_unified
|-- ArduinoOTA @ 1.0
|-- DNSServer @ 1.1.1
|-- ESP8266HTTPClient @ 1.2
|-- ESP8266httpUpdate @ 1.3
|-- ESP8266WebServer @ 1.0
|-- ESP8266WiFi @ 1.0
|-- LittleFS @ 0.1.0
|-- OneWire @ 2.3.7
|-- Ticker @ 1.0
Building in release mode
Retrieving maximum program size .pio\build\nodemcuv2\firmware.elf
Checking size .pio\build\nodemcuv2\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [=====     ]  49.6% (used 40644 bytes from 81920 bytes)
Flash: [=====     ]  53.2% (used 556119 bytes from 1044464 bytes)
Configuring upload protocol...
AVAILABLE: espota, esptool
CURRENT: upload_protocol = espota
Uploading .pio\build\nodemcuv2\firmware.bin
09:00:44 [DEBUG]: Options: {'esp_ip': '192.168.2.59', 'host_ip': '0.0.0.0', 'esp_port': 8266, 'host_port': 43874, 'auth': '', 'image': '.pio\\build\\nodemcuv2\\firmware.bin', 'spiffs': False, 'debug': True, 'progress': True}
09:00:44 [INFO]: Starting on 0.0.0.0:43874
09:00:44 [INFO]: Upload size: 560272
09:00:44 [INFO]: Sending invitation to: 192.168.2.59
Authenticating...FAIL
09:00:44 [ERROR]: Authentication Failed
*** [upload] Error 1
================================================ [FAILED] Took 10.67 seconds ================================================

 *  The terminal process "C:\Users\bjoer\.platformio\penv\Scripts\platformio.exe 'run', '--target', 'upload'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

**Reply by visualapproach** [reply]
Indentation matters

**Reply by dozwin81** [reply]
Sorry, i dont understand what do you mean?

**Reply by visualapproach** [reply]
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/22987250/febc2c80-c753-4331-9fc2-b4bef21f8cb1)

**Reply by dozwin81** [reply]
I'm sorry, which line do I have to do? Or can you please send me the right Ini?

**Reply by visualapproach** [reply]
You already have the file, and you only need to comment out one line and uncomment four, all described in the manual. It says to use ctrl' to avoid indentation errors, just like I mentioned in my first answer above the million lines of text you wanted me to read. Judging from your text in gigantic font it was something about AUTH FAILED, so looking at that part in .ini there is no indentation which I guess is your problem.

**Reply by dozwin81** [reply]
Thanks for your answer, i have find the indent your mean and it works, but whith copy and past your lines from the manuel works not. The space must be two ;-)
